trin-beacon/src/storage.rs

```diff
@@ -0,0 +1,532 @@
+                    .map_err(|err| {
+                        ContentStoreError::Database(format!(
+                            "Error looking up content key: {err:?}"
+                        ))
+                    })?
+                    .is_some();
+                if is_key_available {
+                    return Ok(ShouldWeStoreContent::AlreadyStored);
+                }
+                Ok(ShouldWeStoreContent::Store)
+            }
+            BeaconContentKey::LightClientUpdatesByRange(content_key) => {
+                // Check if any of the periods are available, return AlreadyStored if so otherwise
+                // Store
+                let periods =
+                    content_key.start_period..(content_key.start_period + content_key.count);
+
+                for period in periods {
+                    let is_period_available = self
+                        .lookup_lc_update_period(period)
+                        .map_err(|err| {
+                            ContentStoreError::Database(format!(
+                                "Error looking up lc update period: {err:?}"
+                            ))
+                        })?
+                        .is_some();
+                    if is_period_available {
+                        return Ok(ShouldWeStoreContent::AlreadyStored);
+                    }
+                }
+                Ok(ShouldWeStoreContent::Store)
+            }
+            BeaconContentKey::LightClientFinalityUpdate(content_key) => {
+                match self.cache.get_finality_update(content_key.finalized_slot) {
+                    Some(_) => Ok(ShouldWeStoreContent::AlreadyStored),
+                    None => Ok(ShouldWeStoreContent::Store),
+                }
+            }
+            BeaconContentKey::LightClientOptimisticUpdate(content_key) => {
+                match self.cache.get_optimistic_update(content_key.signature_slot) {
+                    Some(_) => Ok(ShouldWeStoreContent::AlreadyStored),
+                    None => Ok(ShouldWeStoreContent::Store),
+                }
+            }
+        }
+    }
+
+    /// The "radius: concept is not applicable for Beacon network, this is why we always return the
+    /// max radius.
+    fn radius(&self) -> Distance {
+        Distance::MAX
+    }
+}
+
+impl BeaconStorage {
+    pub fn new(config: PortalStorageConfig) -> Result<Self, ContentStoreError> {
+        let metrics = StorageMetricsReporter {
+            storage_metrics: PORTALNET_METRICS.storage(),
+            protocol: ProtocolId::Beacon.to_string(),
+        };
+        let storage = Self {
+            node_data_dir: config.node_data_dir,
+            sql_connection_pool: config.sql_connection_pool,
+            storage_capacity_in_bytes: config.storage_capacity_mb * BYTES_IN_MB_U64,
+            metrics,
+            network: ProtocolId::Beacon,
+            cache: BeaconStorageCache::new(),
+        };
+
+        // Report current storage capacity.
```

Comment:
Yes, you are right, those metrics report the whole storage because we will need also to refactor  `StorageMetricsReporter` to support multiple networks.

I'm happy to leave this for now and create a new issue that will deal with supporting metrics for multiple networks, what do you think?